plugins/services/src/js/containers/service-connection/SDKServiceConnectionEndpointList.js

```diff
@@ -0,0 +1,265 @@
+          >
+            <Icon id="clipboard" size="mini" ref="copyButton" color="grey" />
+          </ClipboardTrigger>
+        </div>
+        {command}
+      </div>
+    );
+  }
+
+  getJSONEndpoint(endpoint) {
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationMapHeading>
+          {endpoint.getName()}
+        </ConfigurationMapHeading>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Address
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            <span>{this.getClipboardTrigger(endpoint.getAddress())}</span>
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            DNS
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getDns())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            VIP
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {this.getClipboardTrigger(endpoint.getVip())}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+      </ConfigurationMapSection>
+    );
+  }
+
+  getFileEndpoint(endpoint) {
+    return (
+      <ConfigurationMapRow key={endpoint.getName()}>
+        <ConfigurationMapLabel>
+          {endpoint.getName()}
+        </ConfigurationMapLabel>
+        <ConfigurationMapValue>
+          <a
+            className="active endpoint-download"
+            download={endpoint.getName()}
+            href={RouterUtil.getResourceDownloadPath(
+              "text/plain",
+              endpoint.getName(),
+              endpoint.getData()
+            )}
+          >
+            <span>
+              <Icon
+                id="download"
+                className="endpoint-download-icon"
+                size="mini"
+              />
+              {" "}
+              Download
+            </span>
+          </a>
+        </ConfigurationMapValue>
+      </ConfigurationMapRow>
+    );
+  }
+
+  getJSONEndpoints(endpoints) {
+    const jsonEndpoints = endpoints.filter(endpoint => {
+      return endpoint.isJSON();
+    });
+
+    if (jsonEndpoints.length === 0) {
+      return null;
+    }
+
+    return jsonEndpoints.map(endpoint => {
+      return this.getJSONEndpoint(endpoint);
+    });
+  }
+
+  getFileEndpointRows(fileEndpoints) {
+    return fileEndpoints.map(endpoint => {
+      return this.getFileEndpoint(endpoint);
+    });
+  }
+
+  getFileEndpoints(endpoints) {
+    const fileEndpoints = endpoints.filter(endpoint => {
+      return !endpoint.isJSON();
+    });
+
+    if (fileEndpoints.length === 0) {
+      return null;
+    }
+
+    return (
+      <ConfigurationMapSection>
+        <ConfigurationMapHeading>
+          Files
+        </ConfigurationMapHeading>
+        {this.getFileEndpointRows(fileEndpoints)}
```

Comment:
_Not_ saying that's wrong, but what's the benefit of extracting `getFileEndpointRows` into it's own function?

Reply:
Deleted the function and moved it in that area